swarm/runtime/shadow_fork.py

```
@@ -222,17 +275,73 @@ def create(self, base_branch: str = "main") -> str:
+            # Try to acquire exclusive lock (non-blocking)
+            if not self._acquire_marker_lock(marker_file):
+                # Lock is held by another process - shadow fork already active
+                marker_file.close()
+                # Read the existing marker content for the error message
+                try:
+                    existing_content = marker_path.read_text().strip()
+                    shadow_branch = existing_content.split('\n')[0].split('=')[1] if '=' in existing_content else "unknown"
+                    raise RuntimeError(
+                        f"Shadow fork already active. Existing shadow: {shadow_branch}. "
+                        "Call cleanup() before creating a new shadow fork."
+                    )
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)